hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/utils/WeightValidationHelper.java

```diff
@@ -0,0 +1,64 @@
+package org.eclipse.hawkbit.repository.jpa.utils;
+
+import org.eclipse.hawkbit.repository.TenantConfigurationManagement;
+import org.eclipse.hawkbit.repository.exception.MultiAssignmentIsNotEnabledException;
+import org.eclipse.hawkbit.repository.exception.NoWeightProvidedInMultiAssignmentModeException;
+import org.eclipse.hawkbit.security.SystemSecurityContext;
+
+/**
+ * Utility class to handle weight validation in Rollout, Auto Assignments, and
+ * Online Assignment.
+ */
+public final class WeightValidationHelper {
+
+    private WeightValidationHelper() {
+        // utility class
+    }
```

Comment:
yet another static utiltiy  :worried:

Reply:
To improve a little bit: I suggest to write the WeightValidator like this:
```
public final class WeightValidationHelper {

    private final TenantConfigurationManagement tenantConfigurationManagement;
    private final SystemSecurityContext systemSecurityContext;

    public WeightValidationHelper(final SystemSecurityContext systemSecurityContext,
            final TenantConfigurationManagement tenantConfigurationManagement) {
        this.systemSecurityContext = systemSecurityContext;
        this.tenantConfigurationManagement = tenantConfigurationManagement;
    }

    // static factory method for better code readability
    public static WeightValidationHelper usingContext(final SystemSecurityContext systemSecurityContext,
            final TenantConfigurationManagement tenantConfigurationManagement) {
        return new WeightValidationHelper (systemSecurityContext, tenantConfigurationManagement);
    }

    public void validate(final List<DeploymentRequest> deploymentRequests) {
        // do stuff
    }

    public void validate(final Rollout rollout) {
        // do stuff

    }

    public void validate(final AutoAssignDistributionSetUpdate update) {
        // do stuff

    }
```
As an result the you can us the validator as a spring bean - or in intermediate step create an instance either in the constructor or directly where needed:
```
    @Override
    @Transactional(isolation = Isolation.READ_COMMITTED)
    public List<DistributionSetAssignmentResult> assignDistributionSets(
            final List<DeploymentRequest> deploymentRequests, final String actionMessage) {
        // validateOnlineAssignment(deploymentRequests);
       WeightValidationHelper.usingContext(systemSecurityContext, tenantConfigurationManagement).validate(deploymentRequests);
        return assignDistributionSets(deploymentRequests, actionMessage, onlineDsAssignmentStrategy);
    }
```

Reply:
Spoke to @vigier and got both the helper classes fixed.